src/Data/Dec/Base.agda

```diff
@@ -61,6 +61,24 @@ open Dec public
 pattern yes p =  true because ofʸ  p
 pattern no ¬p = false because ofⁿ ¬p
 
+dec-elim : {C : Dec P → Type ℓ′}
```

Comment:
Recursion/induction principles, functor laws have names like `elim`, `rec`, `map` etc, it really helps discoverability. Also forces user to explicitly disambiguate via renaming/qualified names.